### PR TITLE
deps: bump SDK to v0.46.6

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
deps: bump Cosmos SDK to v0.46.6 and Tendermint to v0.34.23.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by johnletey** [comment]
@crodriguezvega Would recommend you take a look at https://github.com/cosmos/cosmos-sdk/pull/13918.

We have found a bug in the `x/gov` module since `v0.46.5`.

**Comment by crodriguezvega** [comment]
> @crodriguezvega Would recommend you take a look at [cosmos/cosmos-sdk#13918](https://github.com/cosmos/cosmos-sdk/pull/13918).
> 
> We have found a bug in the `x/gov` module since `v0.46.5`.

Thanks for the heads up, @johnletey!

**Comment by faddat** [comment]
good work to both teams here.  Thanks!

File: e2e/go.mod [hunk 1]
```diff
@@ -5,16 +5,16 @@ go 1.18
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
 require (
-	github.com/cosmos/cosmos-sdk v0.46.4
+	github.com/cosmos/cosmos-sdk v0.46.5
```
Comment (julienrbrt) [inline e2e/go.mod:8]:
```suggestion
	github.com/cosmos/cosmos-sdk v0.46.6
```

**Comment by julienrbrt** [comment]
We've just tagged a 0.46.6, I think you could bump directly to this one.

**Comment by crodriguezvega** [comment]
> We've just tagged a 0.46.6, I think you could bump directly to this one.

Thanks for the quick turnaround, @julienrbrt!

**Review by damiannolan — APPROVED** [review]

**Comment by damiannolan** [comment]
Thank you @johnletey @julienrbrt @crodriguezvega 🙏

**Pull request merged**